plugins/modules/files/archive.py

```diff
@@ -378,7 +376,15 @@ def add_targets(self):
             )
 
     def compare_with_original(self):
-        self.changed |= self.original_size != self.destination_size()
+        if self.original_checksums is None:
+            self.changed |= self.original_size != self.destination_size()
```

Comment:
If `original_checksums is None`, doesn't that mean that the destination file does not exist (or is not readable)? If so, then what is the point of checking whether the original_size (which must be zero) is different from the destination size (which should be greater than zero, right)? Shouldn't we assume the comparison will always result in "different" for that case?

Reply:
It addresses an edge case for archives without `read` access.

Consider the following:
```python
       - community.general.archive:
           path:
             - a_file
             - another_file
           dest: an_archive
           mode: 0220
       - community.general.archive:
           path:
             - a_file
             - another_file
             - a_third_file
           dest: an_archive
           mode: 0220
```
Prior to this PR a change would be detected as the archive size has changed. Checksum comparison alone however would not be able to detect this.

Reply:
Nice catch.